### ISSUE TITLE
Reimplement FactionsDynmap

### ISSUE THREAD

**Issue by ghost** [magnusulf/Factions#15]
Will you modify the FactionDynmap plugin for the version of minecraftu 1.13?

**Comment by magnusulf** [comment]
Yes and no. FactionsDynmap is proprietary and I can't use that source code legally. Although I do have plans to make some Dynmap integration at some point.

**Comment by ghost** [comment]
It might be easier to edit https://dev.bukkit.org/projects/dynmap-factions

**Comment by ghost** [comment]
I did a patch-job and compiled it against Factions3 and the latest Spigot 1.14.2 snapshot: https://github.com/korvyr/Dynmap-Factions

Currently working great. Feel free to take it and make it your own, @magnusulf. :D

**Comment by Uranusak** [comment]
> I did a patch-job and compiled it against Factions3 and the latest Spigot 1.14.2 snapshot: https://github.com/korvyr/Dynmap-Factions
> 
> Currently working great. Feel free to take it and make it your own, @magnusulf. :D

some error in spigot 1.14.3  @korvyr 

```
[05:08:17] [Server thread/INFO]: [Dynmap-Factions] Enabling Dynmap-Factions v0.91-Dev201906301749
[05:08:17] [Server thread/INFO]: [Dynmap-Factions] initializing
[05:08:17] [Server thread/ERROR]: Error occurred while enabling Dynmap-Factions v0.91-Dev201906301749 (Is it up to date?)
java.lang.NullPointerException: null
        at org.dynmap.bukkit.DynmapPlugin.getMarkerAPI(DynmapPlugin.java:981) ~[?:?]
        at org.dynmap.factions.DynmapFactionsPlugin.activate(DynmapFactionsPlugin.java:617) ~[?:?]
        at org.dynmap.factions.DynmapFactionsPlugin.onEnable(DynmapFactionsPlugin.java:605) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[spigot.jar:git-Spigot-935adb3-a0e4747]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:352) [spigot.jar:git-Spigot-935adb3-a0e4747]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:416) [spigot.jar:git-Spigot-935adb3-a0e4747]
        at org.bukkit.craftbukkit.v1_14_R1.CraftServer.enablePlugin(CraftServer.java:460) [spigot.jar:git-Spigot-935adb3-a0e4747]
        at org.bukkit.craftbukkit.v1_14_R1.CraftServer.enablePlugins(CraftServer.java:374) [spigot.jar:git-Spigot-935adb3-a0e4747]
        at net.minecraft.server.v1_14_R1.MinecraftServer.a(MinecraftServer.java:439) [spigot.jar:git-Spigot-935adb3-a0e4747]
        at net.minecraft.server.v1_14_R1.DedicatedServer.init(DedicatedServer.java:258) [spigot.jar:git-Spigot-935adb3-a0e4747]
        at net.minecraft.server.v1_14_R1.MinecraftServer.run(MinecraftServer.java:764) [spigot.jar:git-Spigot-935adb3-a0e4747]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
```

**Comment by TendraCat** [comment]
@korvyr Does this work on version 1.13.2?

**Comment by ghost** [comment]
@TendraTheWolf It should. If it doesn't, let me know and I'll create a version that does.

**Comment by TendraCat** [comment]
@korvyr It dosen't look like its working, unless i'm stupid.

**Comment by ghost** [comment]
@TendraTheWolf I've tossed my compiled jar up on my server so you can grab it and give it a whirl:

~~https://mountainmengaming.com/korvyrwuzhere/Dynmap-Factions-kmod.jar~~
Edit: Updated link: https://github.com/korvyr/Dynmap-Factions/releases/tag/0.83

Let me know how it goes.

**Comment by TMCThomas** [comment]
@korvyr for me it works, thanks! :D 
Do you happen know if it's possible to add more "stats" to this plugin like the official dynmap one? For example the description of a Faction?

**Comment by TendraCat** [comment]
@korvyr Thank you! It worked perfectly. I'm also wondering what TMC said.

**Comment by ghost** [comment]
@TMCThomas @TendraTheWolf Yup, that's all done in the configuration file.

Exempli gratia, here's my config file, note the "infowindow" option specifically:
`# /srv/minecraft/plugins/Dynmap-Factions/config.yml:`
`# Dynmap-Factions configuration`
`update:`
`  period: 300`
`layer:`
`  name: Factions`
`  hidebydefault: false`
`  layerprio: 2`
`  minzoom: 0`
`infowindow: <div class="infowindow"><span style="font-size:120%; font-weight:bold;">%regionname%</span><br/>%description%<br/><br/><span style="font-weight:bold;">Leader:</span><br/>%playerowners%<br/><br/><span style="font-weight:bold;">Members:</span><br/>%playermembers%</div>`
`visibility-by-faction: true`
`regionstyle:`
`  strokeColor: '#FF0000'`
`  strokeOpacity: 0.8`
`  strokeWeight: 3`
`  fillColor: '#FF0000'`
`  fillOpacity: 0.35`
`  homeicon: redflag`
`  boost: false`
`visibleregions: []`
`hiddenregions: []`
`custstyle:`
`  SafeZone:`
`    strokeColor: '#0000FF'`
`    fillColor: '#0000FF'`
`    boost: false`
`  WarZone:`
`    strokeColor: '#404040'`
`    fillColor: '#404040'`
`    boost: false`

~~You can see it in action here: http://mountainmengaming.com:8123~~

**Comment by TMCThomas** [comment]
Thanks that works perfect! It's a shame homes were removed from factions, the flag at the home is the only thing that's missing

**Comment by ghost** [comment]
Yeah, this was very much a patch job, and since Factions3 went to multiple "warps" instead of a single "home" I removed the entire flags function for now to avoid the plugin crashing. I'm hoping someone else will pick it up who has the time to add it back in.

**Comment by amxrmxhdx** [comment]
@korvyr Somehow the link to the compiled jar doesn't work :/ Can you send a new one maybe?

**Comment by ghost** [comment]
@amxrmxhdx I removed it after a while. Let me recompile and upload it to Github and I'll post a permanent link here.

**Comment by ghost** [comment]
And here it is: https://github.com/korvyr/Dynmap-Factions/releases/tag/0.83

**Comment by TMCThomas** [comment]
@korvyr do you happen to have time to update it to 1.15? :)

**Comment by ghost** [comment]
@TMCThomas It should work if this factions plugin hasn't changed since I created it. Are you getting a specific error from the plugin? If so, give me a snippet of the error and I'll go fix it and compile a new version for ya.

**Comment by TMCThomas** [comment]
**EDIT: Nvm! It's my and dynmap's fault, at first it was due to an error of mine, now it's because dynmap doesn't work correctly on 1.15.1, sorry!**

I'm getting the following error:

18.12 00:34:39 [Server] INFO [Dynmap-Factions] Enabling Dynmap-Factions v0.91-Dev201906200550*
18.12 00:34:39 [Server] INFO [Dynmap-Factions] initializing
18.12 00:34:39 [Server] ERROR Error occurred while enabling Dynmap-Factions v0.91-Dev201906200550 (Is it up to date?)
18.12 00:34:39 [Server] INFO java.lang.NullPointerException: null
18.12 00:34:39 [Server] INFO at org.dynmap.bukkit.DynmapPlugin.getMarkerAPI(DynmapPlugin.java:1010) ~[?:?]
18.12 00:34:39 [Server] INFO at org.dynmap.factions.DynmapFactionsPlugin.activate(DynmapFactionsPlugin.java:617) ~[?:?]
18.12 00:34:39 [Server] INFO at org.dynmap.factions.DynmapFactionsPlugin.onEnable(DynmapFactionsPlugin.java:605) ~[?:?]
18.12 00:34:39 [Server] INFO at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[patched_1.15.1.jar:git-Paper-22]
18.12 00:34:39 [Server] INFO at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:338) ~[patched_1.15.1.jar:git-Paper-22]
18.12 00:34:39 [Server] INFO at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:420) ~[patched_1.15.1.jar:git-Paper-22]
18.12 00:34:39 [Server] INFO at org.bukkit.craftbukkit.v1_15_R1.CraftServer.enablePlugin(CraftServer.java:468) ~[patched_1.15.1.jar:git-Paper-22]
18.12 00:34:39 [Server] INFO at org.bukkit.craftbukkit.v1_15_R1.CraftServer.enablePlugins(CraftServer.java:382) ~[patched_1.15.1.jar:git-Paper-22]
18.12 00:34:39 [Server] INFO at net.minecraft.server.v1_15_R1.MinecraftServer.a(MinecraftServer.java:481) ~[patched_1.15.1.jar:git-Paper-22]
18.12 00:34:39 [Server] INFO at net.minecraft.server.v1_15_R1.DedicatedServer.init(DedicatedServer.java:290) ~[patched_1.15.1.jar:git-Paper-22]
18.12 00:34:39 [Server] INFO at net.minecraft.server.v1_15_R1.MinecraftServer.run(MinecraftServer.java:884) ~[patched_1.15.1.jar:git-Paper-22]
18.12 00:34:39 [Server] INFO at java.lang.Thread.run(Thread.java:748) [?:1.8.0_222]
18.12 00:34:39 [Server] INFO [Dynmap-Factions] Disabling Dynmap-Factions v0.91-Dev201906200550

Would be very thankfull for a fixed buildfor 1.15, it's great to see factions on the dynmap!
